packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -857,6 +860,75 @@ export class GarbageCollector implements IGarbageCollector {
 		return this.summaryStateTracker.refreshLatestSummary(result);
 	}
 
+	/**
+	 * Process a GC message.
+	 * @param message - The GC message from the container runtime.
+	 * @param local - Whether it was send by this client.
+	 */
+	public processMessage(message: ContainerRuntimeGCMessage, local: boolean) {
+		switch (message.contents.type) {
+			case GarbageCollectionMessageType.Sweep: {
+				// Delete the nodes whose ids are present in the contents.
+				const sweepMessageContents = message.contents.contents;
+				this.deleteSweepReadyNodes(sweepMessageContents.deletedNodeIds);
+				break;
+			}
+			default: {
+				const compatBehavior = message.compatDetails?.behavior;
+				if (compatBehavior !== "Ignore") {
+					const error = DataProcessingError.create(
+						`Garbage collection message of unknown type ${message.type}`,
+						"processMessage",
+					);
+					this.runtime.closeFn(error);
```

Comment:
Off-topic tangent: I always have mixed feelings about `close(error); throw error`. I wish we could trust callers to do the right thing - close or not.  I mean, we actually can - DeltaManager and DeltaQueue have the "top-level exception handler" logic that closes the container if _anything_ on the process path throws.  Maybe I'll open a PR to remove this paranoid pattern.

Reply:
(feel free to resolve - just thinking out loud)

Reply:
Changed it to throw only and not close container.